### ISSUE TITLE
problem that appears to be related to subsampling

### ISSUE THREAD

**Issue by adriannotimportant** [haraldk/TwelveMonkeys#331]
Hi Harald,

I'm getting the following error for the attached file:

[test.zip](https://github.com/haraldk/TwelveMonkeys/files/837362/test.zip)

````
java.lang.ArrayIndexOutOfBoundsException
	at java.lang.System.arraycopy(Native Method)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.readStripTileData(TIFFImageReader.java:1518)
	at com.twelvemonkeys.imageio.plugins.tiff.TIFFImageReader.read(TIFFImageReader.java:934)
```` 
I would appreciate any feedback.

Cheers,
adrian

**Comment by haraldk** [comment]
Hi Adrian,

Thanks for reporting!

The title indicates you are reading the image subsampled. Can you please also attach the code you use to expose the problem? I can read the image fine here without subsampling.

There are some known issues related to subsampling, see #289. As your image is 1 bit/pixel, that issue probably also affects you. 

-- 
Harald K

**Comment by adriannotimportant** [comment]
Hi Harald,


Thank you for the prompt reply.
Yes, the image works well without subsampling. The code is quite convoluted because we treat many images using the same frameworkand we have dedicated data structures for data sources to deal with the issue of data sources that don't come from files. It also needs a heap of libraries.
It's probably better to include a screenshot where you could see the parameters values.


Cheers,
Adrian
![stack trace](https://cloud.githubusercontent.com/assets/8119004/23878641/bd03aaa2-089c-11e7-87f4-8cdcf100fd23.png)


Probably the best check is to read the file with subsampling as the proof is in the Pudding as they say.


Cheers,
Adrian

**Comment by haraldk** [comment]
Okay,

So I only need to specify `param.setSourceSubSampling(3, 3, 0, 0)` then. ;-)
Anyway, this is the same issue as #289 only for TIFF, and it's a known issue. Subsampling does not work properly for < 8 bits/sample. I will do my best to fix this for the 3.4 release. 

-- 
Harald K
